File: src/android/FirebasePluginMessageReceiver.java
package org.apache.cordova.firebase;

import com.google.firebase.messaging.RemoteMessage;

public abstract class FirebasePluginMessageReceiver {

    public FirebasePluginMessageReceiver() {
        FirebasePluginMessageReceiverManager.register(this);
    }

    /**
     * Concrete subclasses should override this and return true if they handle the received message.
     *
     * @param remoteMessage
     * @return true if the received message was handled by the receiver so should not be handled by FirebasePlugin.
     */
    public abstract boolean onMessageReceived(RemoteMessage remoteMessage);
}

File: src/ios/FirebasePluginMessageReceiver.m
#import "FirebasePluginMessageReceiver.h"
#import "FirebasePluginMessageReceiverManager.h"

@implementation FirebasePluginMessageReceiver

- (id) init {
    [FirebasePluginMessageReceiverManager register:self];
    return self;
}

// Concrete subclasses should override this and return true if they handle the received message.
- (bool) sendNotification:(NSDictionary *)userInfo {
    NSAssert(false, @"You cannot call sendNotification on the FirebasePluginMessageReceiver class directly. Instead, you must override it using a subclass.");
    return false;
}

@end
